frontend/public/components/namespace.jsx

```diff
@@ -82,11 +92,41 @@ import {
 import { removeQueryArgument } from './utils/router';
 import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
 
+import NamespaceDropDown from './namespace-dropdown';
+import { isCurrentUser, isOtherUser, isSystemNamespace } from './factory/table-filters';
+
 const getModel = (useProjects) => (useProjects ? ProjectModel : NamespaceModel);
 const getDisplayName = (obj) =>
   _.get(obj, ['metadata', 'annotations', 'openshift.io/display-name']);
+
 const CREATE_NEW_RESOURCE = '#CREATE_RESOURCE_ACTION#';
 
+export const REQUEST_FILTER = { me: 'me', user: 'user', system: 'system' };
+
+const getFilters = () => [
+  {
+    filterGroupName: 'Requester',
+    type: 'requester',
+    reducer: (namespace) => {
+      const name = namespace.metadata?.name;
+      const requestor = namespace.metadata?.annotations['openshift.io/requester'];
+      if (isCurrentUser(requestor)) {
+        return REQUEST_FILTER.me;
+      }
+      if (isOtherUser(requestor, name)) {
+        return REQUEST_FILTER.user;
+      }
+      if (isSystemNamespace({ title: name })) {
+        return REQUEST_FILTER.system;
+      }
+    },
+    items: [
+      { id: REQUEST_FILTER.me, title: 'Me' },
+      { id: REQUEST_FILTER.user, title: 'User' },
+      { id: REQUEST_FILTER.system, title: 'System' },
```

Comment:
Should these titles be translated?

Reply:
Yes it should.  Thanks for catching this.  Changes on next commit.